blockchain/chain_sync/src/bucket.rs

```diff
@@ -68,6 +80,36 @@ impl SyncBucketSet {
         // Return the heaviest tipset bucket
         vals.iter().max_by_key(|b| b.weight()).cloned()
     }
+    /// Updates SyncBucketSet by removing specified SyncBucket
+    pub(crate) fn remove(&mut self, ts_bucket: &SyncBucket) {
+        let vals: Vec<SyncBucket> = self
+            .buckets
+            .clone()
```

Comment:
```suggestion
```

Reply:
Need to clone, the compiler complains otherwise.

Reply:
yeah didn't see all you were doing, replace the function with this instead:
```
        self.buckets.retain(|b| b != ts_bucket);
```
instead of cloning and full tipset and filtering